File: rest/src/main/java/ru/gitpub/rosatom/domain/repos/TaskRepository.java
package ru.gitpub.rosatom.domain.repos;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;
import ru.gitpub.rosatom.domain.entities.Task;

@Repository
public interface TaskRepository extends JpaRepository<Task, Long>, JpaSpecificationExecutor<Task> {

    @Query("select t from Task t where t.parentId is null")
    List<Task> findAll();

    List<Task> findAllByParentId(Long parentId);
}

File: app/RosAtomTaskTracker/app/src/main/java/com/gitpub/rosatomtasktracker/service/repository/tasks/TaskRepository.kt
package com.gitpub.rosatomtasktracker.service.repository.tasks

import com.gitpub.rosatomtasktracker.model.domain.Task
import com.gitpub.rosatomtasktracker.model.util.Result

interface TaskRepository {

    suspend fun getTaskById(id: Long): Result<Task>

    suspend fun getTasks(): Result<List<Task>>
}
